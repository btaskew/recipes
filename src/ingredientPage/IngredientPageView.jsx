import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

import {Alert} from 'react-bootstrap';
import Page from 'components/Page';
import RecipesList from 'components/RecipesList';

class IngredientPageView extends PureComponent {
    async componentDidMount() {
        await this.props.getRecipes(this.props.ingredient);
    }

    render() {
        return (
            <Page heading={`Recipes using "${this.props.ingredient}"`} loading={this.props.loading}>
                {this.props.noResult && (
                    <Alert bsStyle="info">No recipes found for that ingredient</Alert>
                )}

                {this.props.recipes && <RecipesList recipes={this.props.recipes} />}
            </Page>
        );
    }
}

IngredientPageView.propTypes = {
    ingredient: PropTypes.string.isRequired,
    getRecipes: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    noResult: PropTypes.bool,
    recipes: PropTypes.array
};
IngredientPageView.defaultProps = {
    noResult: false,
    recipes: []
};

export default IngredientPageView;
