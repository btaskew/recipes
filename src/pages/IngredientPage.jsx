import React, {Component} from 'react';

import Recipe from 'scripts/Recipe';

import Page from 'components/utils/Page';
import RecipesList from 'components/RecipesList';

class IngredientPage extends Component {
    constructor(props) {
        super(props);
        this.state = {recipes: null};
    }

    componentDidMount() {
        const recipes = Recipe.findByIngredient(this.props.match.params.name);
        this.setState({recipes});
    }

    render() {
        return (
            <Page
                heading={`Recipes using "${this.props.match.params.name}"`}
                homeIcon
            >
                {this.state.recipes && (
                    <RecipesList recipes={this.state.recipes} />
                )}
            </Page>
        );
    }
}

export default IngredientPage;
