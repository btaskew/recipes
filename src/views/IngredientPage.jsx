import React from 'react';

import Page from 'components/utils/Page';
import RecipesList from 'components/RecipesList';
import GetRecipes from 'containers/GetRecipes';

function IngredientPage(props) {
    return (
        <GetRecipes
            name={props.match.params.name}
            render={state => (
                <Page
                    heading={`Recipes using "${props.match.params.name}"`}
                    error={state.error}
                    noResult={state.noResult}
                >
                    {state.recipes && (
                        <RecipesList recipes={state.recipes} />
                    )}
                </Page>
            )}
        />
    );
}

export default IngredientPage;
