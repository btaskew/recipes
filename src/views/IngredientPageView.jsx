import React, {PureComponent} from 'react';

import {Alert} from 'react-bootstrap';
import Page from 'components/utils/Page';
import RecipesList from 'components/RecipesList';

class IngredientPageView extends PureComponent {
    componentDidMount() {
        this.props.getRecipes(this.props.ingredient);
    }

    render() {
        return (
            <Page
                heading={`Recipes using "${this.props.ingredient}"`}
                loading={this.props.loading}
            >
                {this.props.noResult && (
                    <Alert bsStyle="info">
                        No recipes found for that ingredient
                    </Alert>
                )}

                {this.props.recipes && (
                    <RecipesList recipes={this.props.recipes} />
                )}
            </Page>
        );
    }
}

export default IngredientPageView;
