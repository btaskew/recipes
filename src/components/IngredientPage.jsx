import React from 'react';
import RecipeSearch from 'containers/RecipeSearch';
import IngredientPageView from 'views/IngredientPageView';

function IngredientPage(props) {
    return (
        <RecipeSearch
            render={(state, recipeSearch) => (
                <IngredientPageView
                    {...state}
                    ingredient={props.match.params.name}
                    getRecipes={recipeSearch}
                />
            )}
        />
    );
}

export default IngredientPage;
