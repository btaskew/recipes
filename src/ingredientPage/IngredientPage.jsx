import React from 'react';
import PropTypes from 'prop-types';

import RecipeSearch from 'containers/RecipeSearch';
import IngredientPageView from 'ingredientPage/IngredientPageView';

function IngredientPage(props) {
    return (
        <RecipeSearch
            render={(state, recipeSearch) => (
                <IngredientPageView
                    {...state}
                    ingredient={props.match.params.name}
                    getRecipes={recipeSearch}
                />
            )}
        />
    );
}

IngredientPage.propTypes = {
    match: PropTypes.object
};
IngredientPage.defaultProps = {
    match: {
        params: {
            name: ''
        }
    }
};

export default IngredientPage;
