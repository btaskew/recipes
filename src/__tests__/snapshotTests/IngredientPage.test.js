import React from 'react';
import renderer from 'react-test-renderer';

import IngredientPage from 'ingredientPage/IngredientPage';
import IngredientPageView from 'ingredientPage/IngredientPageView';
import {MemoryRouter} from 'react-router';

describe('Ingredient page', () => {
    it('renders correctly', () => {
        const ingredientPage = renderer
            .create(
                <MemoryRouter>
                    <IngredientPage />
                </MemoryRouter>
            )
            .toJSON();
        expect(ingredientPage).toMatchSnapshot();
    });
});

describe('Ingredient page view', () => {
    it('renders loading correctly', () => {
        const ingredientPage = renderer
            .create(
                <MemoryRouter>
                    <IngredientPageView loading={true} ingredient="Test" getRecipes={() => {}} />
                </MemoryRouter>
            )
            .toJSON();
        expect(ingredientPage).toMatchSnapshot();
    });

    it('renders when no result correctly', () => {
        const ingredientPage = renderer
            .create(
                <MemoryRouter>
                    <IngredientPageView
                        loading={false}
                        recipes={null}
                        ingredient="Test"
                        getRecipes={() => {}}
                    />
                </MemoryRouter>
            )
            .toJSON();
        expect(ingredientPage).toMatchSnapshot();
    });

    it('renders with recipes correctly', () => {
        const testRecipes = [
            {
                id: 1,
                name: 'Recipe 1'
            },
            {
                id: 2,
                name: 'Recipe 2'
            }
        ];

        const ingredientPage = renderer
            .create(
                <MemoryRouter>
                    <IngredientPageView
                        loading={false}
                        recipes={testRecipes}
                        ingredient="Test"
                        getRecipes={() => {}}
                    />
                </MemoryRouter>
            )
            .toJSON();
        expect(ingredientPage).toMatchSnapshot();
    });
});
