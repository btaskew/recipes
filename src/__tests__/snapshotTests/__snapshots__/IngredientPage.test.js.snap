// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredient page renders correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Recipes using ""
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="alert alert-info"
      role="alert"
    >
      No recipes found for that ingredient
    </div>
  </div>,
]
`;

exports[`Ingredient page view renders loading correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Recipes using "Test"
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="alert alert-info"
      role="alert"
    >
      Loading...
    </div>
  </div>,
]
`;

exports[`Ingredient page view renders when no result correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Recipes using "Test"
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="alert alert-info"
      role="alert"
    >
      No recipes found for that ingredient
    </div>
    <div
      className="alert alert-info"
      role="alert"
    >
      No recipes provided
    </div>
  </div>,
]
`;

exports[`Ingredient page view renders with recipes correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Recipes using "Test"
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="list-group"
    >
      <a
        className="list-group-item"
        href="/recipe/1"
        onClick={[Function]}
        style={undefined}
      >
        Recipe 1
      </a>
      <a
        className="list-group-item"
        href="/recipe/2"
        onClick={[Function]}
        style={undefined}
      >
        Recipe 2
      </a>
    </div>
  </div>,
]
`;
